funsor/numpy.py

```diff
@@ -11,17 +17,11 @@
 # Register Ops
 ################################################################################
 
-try:
-    from scipy.special import expit as _sigmoid
-except ImportError:
-    def _sigmoid(x):
-        return 1 / (1 + np.exp(-x))
-
 # take care of scalar numpy objects
-array = (np.ndarray, np.generic)
+array = (onp.ndarray, onp.generic, UnshapedArray, DeviceArray)
```

Comment:
I would like to defer tests for UnshapedArray to a later PR (if there is some problems when applying `jax.jit`).